app/Http/Requests/DestroyPageRequest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Http\Requests\Request;
+use App\Models\Role;
+use Auth;
+
+class DestroyPageRequest extends Request
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        $user = Auth::user();
+        return $user && $user->hasRole(Role::ROLE_ADMIN);
```

Comment:
The `authorize` function is part of the interface for a Form Request object, so it needs to be present. As for checking if a user is an admin or not, that could probably be wrapped, but at the moment it's just two lines, so I didn't see a huge need for that.